plugin/evm/vm_warp_test.go

```diff
@@ -0,0 +1,227 @@
+			return ids.Empty, nil
+		},
+		GetValidatorSetF: func(ctx context.Context, height uint64, subnetID ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+			return map[ids.NodeID]*validators.GetValidatorOutput{
+				nodeID1: {
+					NodeID:    nodeID1,
+					PublicKey: blsPublicKey1,
+					Weight:    50,
+				},
+				nodeID2: {
+					NodeID:    nodeID2,
+					PublicKey: blsPublicKey2,
+					Weight:    50,
+				},
+			}, nil
+		},
+	}
+
+	signersBitSet := set.NewBits()
+	signersBitSet.Add(0)
+	signersBitSet.Add(1)
+
+	warpSignature := &avalancheWarp.BitSetSignature{
+		Signers: signersBitSet.Bytes(),
+	}
+
+	blsAggregatedSignatureBytes := bls.SignatureToBytes(blsAggregatedSignature)
+	copy(warpSignature.Signature[:], blsAggregatedSignatureBytes)
+
+	signedMessage, err := avalancheWarp.NewMessage(
+		unsignedMessage,
+		warpSignature,
+	)
+	require.NoError(t, err)
+
+	getWarpMsgInput, err := warp.PackGetVerifiedWarpMessage()
+	require.NoError(t, err)
+	signedTx1, err := types.SignTx(
+		types.NewTx(&types.DynamicFeeTx{
+			ChainID:   vm.chainConfig.ChainID,
+			Nonce:     1,
+			To:        &warp.Module.Address,
+			Gas:       1_000_000,
+			GasFeeCap: big.NewInt(225 * params.GWei),
+			GasTipCap: big.NewInt(params.GWei),
+			Value:     common.Big0,
+			Data:      getWarpMsgInput,
+			AccessList: types.AccessList{
+				types.AccessTuple{
+					Address:     warp.ContractAddress,
+					StorageKeys: byteUtils.BytesToHashSlice(warp.PackPredicate(signedMessage.Bytes())),
+				},
+			},
+		}),
+		types.LatestSignerForChainID(vm.chainConfig.ChainID),
+		testKeys[0],
+	)
+	require.NoError(t, err)
+	errs = vm.txPool.AddRemotesSync([]*types.Transaction{signedTx1})
+	for i, err := range errs {
+		if err != nil {
+			t.Fatalf("Failed to add tx at index %d: %s", i, err)
+		}
+	}
+	vm.clock.Set(vm.clock.Time().Add(2 * time.Second))
+
+	<-issuer
+
+	// Build, verify, and accept block with valid proposer context.
+	validProposerCtx := &block.Context{
+		PChainHeight: 10,
+	}
+	block2, err := vm.BuildBlockWithContext(context.Background(), validProposerCtx)
+	require.NoError(t, err)
+
+	block2VerifyWithCtx, ok := block2.(block.WithVerifyContext)
+	require.True(t, ok)
+	shouldVerifyWithCtx, err := block2VerifyWithCtx.ShouldVerifyWithContext(context.Background())
+	require.NoError(t, err)
+	require.True(t, shouldVerifyWithCtx)
+	require.NoError(t, block2VerifyWithCtx.VerifyWithContext(context.Background(), validProposerCtx))
+	require.Equal(t, choices.Processing, block2.Status())
+	require.NoError(t, vm.SetPreference(context.Background(), block2.ID()))
+	require.NoError(t, block2.Accept(context.Background()))
+
+	ethBlock := block2.(*chain.BlockWrapper).Block.(*Block).ethBlock
+	verifiedMessageReceipts := vm.blockChain.GetReceiptsByHash(ethBlock.Hash())
+	require.Len(t, verifiedMessageReceipts, 1)
+	verifiedMessageTxReceipt := verifiedMessageReceipts[0]
+	require.Equal(t, types.ReceiptStatusSuccessful, verifiedMessageTxReceipt.Status)
```

Comment:
TODO: add more thorough testing across different valid/invalid signatures and using `BuildWithContext/VerifyWithContext`

Reply:
We should test at least the following cases:
- non-zero number of signatures, but does not meet the threshold
- 100% of validators sign
- no signatures
- outdated signatures (later P-Chain height where validator set has changed)
- `VerifyWithContext` on an already verified block with a new valid/invalid context for the predicate